### PR TITLE
[KYUUBI #1800] Remove oom hook

### LINKED ISSUE #1800: [Bug] Exit spark engine if spark context is stopped
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

If the spark engine get OOM issue or underlying bug, the SparkContext will stop itself. However the engine jvm is still alived and it become a zombie process.

We should aware the status of SparkContext and stop the spark engine if it's already stopped.

The error msg:
```

(No active SparkContext.)

        at org.apache.kyuubi.KyuubiSQLException$.apply(KyuubiSQLException.scala:69)
        at org.apache.kyuubi.KyuubiSQLException$.apply(KyuubiSQLException.scala:73)
        at org.apache.kyuubi.engine.spark.session.SparkSQLSessionManager.openSession(SparkSQLSessionManager.scala:93)
        at org.apache.kyuubi.service.AbstractBackendService.openSession(AbstractBackendService.scala:45)
        at org.apache.kyuubi.service.ThriftBinaryFrontendService.getSessionHandle(ThriftBinaryFrontendService.scala:199)
        at org.apache.kyuubi.service.ThriftBinaryFrontendService.OpenSession(ThriftBinaryFrontendService.scala:208)
        at org.apache.hive.service.rpc.thrift.TCLIService$Processor$OpenSession.getResult(TCLIService.java:1497)
        at org.apache.hive.service.rpc.thrift.TCLIService$Processor$OpenSession.getResult(TCLIService.java:1482)
        at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38)
        at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
        at org.apache.kyuubi.service.authentication.TSetIpAddressProcessor.process(TSetIpAddressProcessor.scala:36)
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:310)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Cannot call methods on a stopped SparkContext.
```

### Affects Version(s)

1.3.1/1.4.0

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

closes #1800

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
> If the spark engine get OOM issue or underlying bug

can you describe the `underlying bug`?

**Comment by yaooqinn** [comment]
If this is actually only an OOM issue, I notice that hive has removed the OOM hook and use jvm oom kill instead, and I suggest we follow it

**Comment by ulysses-you** [comment]
here is the key log
```
22/01/19 00:04:41 WARN ManagedSelector: java.lang.OutOfMemoryError: Java heap space
22/01/19 00:04:41 WARN ManagedSelector: java.lang.OutOfMemoryError: Java heap space
22/01/19 00:04:41 WARN DefaultPromise: An exception was thrown by org.apache.spark.network.server.TransportRequestHandler$$Lambda$1378/1835058533.operationComplete()
java.lang.NoClassDefFoundError: Could not initialize class org.apache.commons.lang3.builder.ToStringBuilder
        at org.apache.spark.network.protocol.RpcResponse.toString(RpcResponse.java:84)
        at java.util.Formatter$FormatSpecifier.printString(Formatter.java:2886)
        at java.util.Formatter$FormatSpecifier.print(Formatter.java:2763)
        at java.util.Formatter.format(Formatter.java:2520)
        at java.util.Formatter.format(Formatter.java:2455)
        at java.lang.String.format(String.java:2940)
        at org.apache.spark.network.server.TransportRequestHandler.lambda$respond$1(TransportRequestHandler.java:273)
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490)

22/01/19 00:04:41 ERROR Utils: uncaught error in thread spark-listener-group-shared, stopping SparkContext
```

**Comment by ulysses-you** [comment]
I see https://github.com/apache/hive/pull/2383. I'm fine to remove oom hook.

**Comment by yaooqinn** [comment]
> I see [apache/hive#2383](https://github.com/apache/hive/pull/2383). I'm fine to remove oom hook.

+1 for this

File: kyuubi-common/src/main/scala/org/apache/kyuubi/util/ExecutorPool.scala [hunk 1]
```diff
@@ -17,37 +17,18 @@
 
 package org.apache.kyuubi.util
 
-import java.util.concurrent.{Future, SynchronousQueue, ThreadPoolExecutor, TimeUnit}
+import java.util.concurrent.{SynchronousQueue, ThreadPoolExecutor, TimeUnit}
 
-case class ExecutorPoolCaptureOom(
+case class ExecutorPool(
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/util/ExecutorPool.scala:22]: seem that we can remove it completely

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
it conflicts 1.4, can you backport?

**Comment by ulysses-you** [comment]
thank you @yaooqinn created https://github.com/apache/incubator-kyuubi/pull/1810

**Comment by hzxiongyinke** [comment]
good job

**Pull request closed without merging**